### PR TITLE
feature: Changes to support remote schema retrieval for task types (question-answering, fill-mask) and added e2e tests for both local and remote hf schema logic. 

### PR DESCRIPTION
**Pull request by ShailavTaneja**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

tox -e py310 -- -v -s tests/integ/sagemaker/serve/test_schema_builder.py::test_model_builder_happy_path_with_task_provided_remote_schema_mode
[Output](https://paste.amazon.com/show/shailav/1712252727) 

tox -e py310 -- -v -s tests/unit/sagemaker/serve/builder/test_model_builder.py
[Output](https://paste.amazon.com/show/shailav/1712253071)

tox -e py310 -- -v -s tests/unit/sagemaker/serve/utils/test_task.py
[Output](https://paste.amazon.com/show/shailav/1712253462)

tox -e py310 -- -v -s tests/integ/sagemaker/serve/test_schema_builder.py::test_model_builder_happy_path_with_task_provided_local_schema_mode
[Output](https://paste.amazon.com/show/shailav/1712259866)

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/serve/builder/model_builder.py [hunk 1]
```diff
@@ -64,6 +64,7 @@
 )
 from sagemaker.workflow.entities import PipelineVariable
 from sagemaker.huggingface.llm_utils import get_huggingface_model_metadata
+from sagemaker_schema_inference_artifacts.huggingface import remote_schema_retriever
```
Comment (samruds) [inline src/sagemaker/serve/builder/model_builder.py:67]: Move this import to be on line 718


            try:
               from sagemaker_schema_inference_artifacts.huggingface import remote_schema_retriever
                sample_inputs, sample_outputs = task.retrieve_local_schemas(model_task)
            except ValueError:

**Comment by ShailavTaneja** [comment]
_cc @samruds for review._

File: src/sagemaker/serve/builder/model_builder.py (hunk 1 above)
Comment (ShailavTaneja) [reply src/sagemaker/serve/builder/model_builder.py]: done

File: tests/integ/sagemaker/serve/test_schema_builder.py [hunk 2]
```diff
@@ -92,37 +94,110 @@ def test_model_builder_happy_path_with_only_model_id_question_answering(
...
-    reason="Testing Schema Builder Simplification feature",
+    reason="Testing Schema Builder Simplification feature - Local Schema",
 )
 @pytest.mark.parametrize(
-    "model_id, task_provided",
+    "model_id, task_provided, instance_type_provided, container_startup_timeout",
     [
-        ("bert-base-uncased", "fill-mask"),
-        ("bert-large-uncased-whole-word-masking-finetuned-squad", "question-answering"),
+        ("distilbert/distilbert-base-uncased-finetuned-sst-2-english", "text-classification", "ml.m5.xlarge", None),
```
Comment (samruds) [inline tests/integ/sagemaker/serve/test_schema_builder.py:114]: Thanks for increasing test coverage and adding thorough test cases

**Review by samruds — APPROVED** [review]

**Comment by samruds** [comment]
```
Run flake and pylint

tox -e pylint -- src/

or

tox -e black-format -- src/

or

tox -e flake8 -- src/

or 

tox -e docstyle -- src/

or

tox -e twine -- src/ 
```

File: src/sagemaker/serve/builder/model_builder.py [hunk 3]
```diff
@@ -714,10 +714,20 @@ def _schema_builder_init(self, model_task: str):
...
         """
         try:
-            sample_inputs, sample_outputs = task.retrieve_local_schemas(model_task)
+            try:
+                sample_inputs, sample_outputs = task.retrieve_local_schemas(model_task)
+            except ValueError:
+                # samples could not be loaded locally, try to fetch it from sagemaker_schema_inference_artifacts
+                from sagemaker_schema_inference_artifacts.huggingface import remote_schema_retriever
+                if model_task in ("text-to-image", "automatic-speech-recognition"):
+                    logger.warning(f"HF SchemaBuilder for {model_task} is in beta mode, and is not guaranteed to work "
```
Comment (samruds) [inline src/sagemaker/serve/builder/model_builder.py:723]: Have we added test for this gating logic?

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run all

**Comment by samruds** [comment]
@mufaddal-rohawala can you help run the checks here

**Pull request closed without merging**